pkg/deploy/controller/deployment/controller.go

```diff
@@ -106,14 +106,35 @@ func (c *DeploymentController) Handle(deployment *kapi.ReplicationController) er
 			break
 		}
 
-		// If the pod already exists, it's possible that a previous CreatePod
-		// succeeded but the deployment state update failed and now we're re-
-		// entering. Ensure that the pod is the one we created by verifying the
-		// annotation on it, and throw a retryable error.
-		if deployerErr != nil && !kerrors.IsNotFound(deployerErr) {
+		switch {
+		case kerrors.IsNotFound(deployerErr):
+			if _, ok := deployment.Annotations[deployapi.DeploymentIgnorePodAnnotation]; ok {
+				return nil
+			}
+
+			// Generate a deployer pod spec.
+			deployerPod, err := c.makeDeployerPod(deployment)
+			if err != nil {
+				return fatalError(fmt.Sprintf("couldn't make deployer pod for %s: %v", deployutil.LabelForDeployment(deployment), err))
+			}
+			// Create the deployer pod.
+			deploymentPod, err := c.pn.Pods(deployment.Namespace).Create(deployerPod)
+			// Retry on error.
+			if err != nil {
+				return actionableError(fmt.Sprintf("couldn't create deployer pod for %s: %v", deployutil.LabelForDeployment(deployment), err))
+			}
+			updatedAnnotations[deployapi.DeploymentPodAnnotation] = deploymentPod.Name
+			nextStatus = deployapi.DeploymentStatusPending
+			glog.V(4).Infof("Created deployer pod %s for deployment %s", deploymentPod.Name, deployutil.LabelForDeployment(deployment))
+
+		case deployerErr != nil:
+			// If the pod already exists, it's possible that a previous CreatePod
+			// succeeded but the deployment state update failed and now we're re-
+			// entering. Ensure that the pod is the one we created by verifying the
+			// annotation on it, and throw a retryable error.
 			return fmt.Errorf("couldn't fetch existing deployer pod for %s: %v", deployutil.LabelForDeployment(deployment), deployerErr)
-		}
-		if deployerErr == nil && deployer != nil {
+
+		default: /* deployerErr == nil */
```

Comment:
I was redundant. It will always be non-nil if deployerErr is nil: https://github.com/openshift/origin/blob/b19839883ebd3f22d9347feff63af26d6b43291e/pkg/deploy/controller/deployment/factory.go#L220